Applications/Resnet/jni/cifar_dataloader.cpp

```diff
@@ -93,13 +118,65 @@ void RandomDataLoader::next(float **input, float **label, bool *last) {
 }
 
 Cifar100DataLoader::Cifar100DataLoader(const std::string &path, int batch_size,
-                                       int splits) {
-  /// NYI!
+                                       int splits) :
+  batch(batch_size),
+  current_iteration(0),
+  file(path, std::ios::binary | std::ios::ate) {
+  constexpr char error_msg[] = "failed to create dataloader, reason: ";
+
+  NNTR_THROW_IF(!file.good(), std::invalid_argument)
+    << error_msg << " Cannot open file";
+
+  auto pos = file.tellg();
+  NNTR_THROW_IF((pos % Cifar100DataLoader::SampleSize != 0),
+                std::invalid_argument)
+    << error_msg << " Given file does not align with the format";
+
+  auto data_size = pos / (Cifar100DataLoader::SampleSize * splits);
+  idxes = std::vector<unsigned int>(data_size);
+  std::cout << "path: " << path << '\n';
+  std::cout << "data_size: " << data_size << '\n';
+  std::iota(idxes.begin(), idxes.end(), 0);
+  std::shuffle(idxes.begin(), idxes.end(), rng);
+
+  iteration_per_epoch = data_size / batch_size;
```

Comment:
This code ignores the last batch if data_size is not a multiple of batch_size. If this is intended, please add as a comment.

Reply:
Done